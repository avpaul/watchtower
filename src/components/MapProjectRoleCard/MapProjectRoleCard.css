.role-card__grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(303px, 1fr));
  grid-row-gap: 15px;
  grid-column-gap: 9px;
}
