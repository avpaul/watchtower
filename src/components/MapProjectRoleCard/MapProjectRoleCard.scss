@import '../../assets/sass/abstract/variables';

.role-card__grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(303px, 1fr));
  grid-row-gap: 15px;
  grid-column-gap: 9px;
}
.role-grid__count {
  color: #333e44;
  font-family: $font-primary;
  font-size: 24px;
  letter-spacing: -1.1px;
  line-height: 31px;
  margin-bottom: 3px;
}
#title {
  margin-bottom: 20px;
}
#title > span {
  height: 18px;
  width: 245px;
  color: rgba(52, 76, 90, 0.62);
  font-family: $font-primary;
  font-size: 14px;
  line-height: 18px;
}
.role-grid__add {
  border-radius: 5px;
  background-color: #10a36d;
  box-shadow: 0 2px 3px 0 rgba(0, 0, 0, 0.11);
  color: #ffffff;
  font-family: $font-primary;
  font-size: 14px;
  font-weight: bold;
  letter-spacing: 0.58px;
  line-height: 18px;
  padding: 10px 20px;
  outline: none !important;
  float: right;
}
.role-grid__add:hover {
  text-decoration: none;
  color: #ffffff;
}
.project-card__grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(303px, 1fr));
  grid-row-gap: 15px;
  grid-column-gap: 9px;
}
.no-roles-certifications {
  font-size: 24px;
  color: #333e44;
  padding: 15px;
}
