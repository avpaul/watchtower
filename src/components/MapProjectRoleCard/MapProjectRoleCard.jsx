import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './MapProjectRoleCard.scss';
import Card from '../RoleAndCertificationCard';
import arrayOfObjectsSorter from '../../utils/sortArray';
import EmptyDashboard from '../WorkInProgress/WorkInProgress';

const renderButtons = type => (
  <div className="col-3">
    {type === 'role' ? (
      <button
        type="button"
        data-toggle="modal"
        data-target="#addRoleModal"
        className="role-grid__add"
      >
        NEW ROLE
      </button>
    ) : (
      <button
        type="button"
        data-toggle="modal"
        data-target="#addCertificationModal"
        className="role-grid__add"
      >
        NEW CERTIFICATION
      </button>
    )}
  </div>
);

const renderHeader = (type, roleData) => (
  <Fragment>
    <div className="col-9" id="title">
      <p className="role-grid__count">
        {type === 'role' ? 'Roles' : 'Certifications'}
      </p>
      <span>
        {roleData.length} Vacant, {roleData.length} Active
      </span>
    </div>
    {renderButtons(type)}
  </Fragment>
);

const MapProjectRoleCard = ({
  roleData,
  type,
  fetchActiveEngineers,
  loading,
  activeEngineers,
  setDeleteTarget,
  setCertificationOnFocus
}) => (
  <div>
    <div className="row">
      {renderHeader(type, roleData)}
      {roleData.length === 0 ? (
        <div className="no-roles-certifications">
          {type === 'role' ? (
            <EmptyDashboard title="Welcome, please create the first Role!" />
          ) : (
            <EmptyDashboard title="Welcome, please create the first Certification!" />
          )}
        </div>
      ) : (
        roleData.sort(arrayOfObjectsSorter('name')).map(role => (
          <div className="col-4 mb-4" key={role.id}>
            <Card
              cardProps={{
                details: role,
                fetcher: fetchActiveEngineers,
                loading,
                activeParticipants: activeEngineers,
                type
              }}
              focusRole={setDeleteTarget}
              setCertificationOnFocus={setCertificationOnFocus}
            />
          </div>
        ))
      )}
    </div>
  </div>
);

MapProjectRoleCard.propTypes = {
  activeEngineers: PropTypes.oneOfType([
    PropTypes.instanceOf(Array),
    PropTypes.shape()
  ]),
  type: PropTypes.string.isRequired,
  setCertificationOnFocus: PropTypes.func.isRequired,
  fetchActiveEngineers: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  roleData: PropTypes.instanceOf(Array),
  setDeleteTarget: PropTypes.func.isRequired
};
MapProjectRoleCard.defaultProps = {
  roleData: [],
  activeEngineers: []
};

export default MapProjectRoleCard;
