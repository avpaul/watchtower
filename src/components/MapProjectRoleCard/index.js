import MapProjectRoleCard from './MapProjectRoleCard';

export default MapProjectRoleCard;
