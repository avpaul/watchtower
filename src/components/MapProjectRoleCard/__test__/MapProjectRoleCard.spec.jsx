import React from 'react';
import { shallow } from 'enzyme';
import MapProjectRoleCard from '../MapProjectRoleCard';

const role = {
  id: 1,
  name: 'test role',
  description: 'test role description',
  active_engineers_count: 1,
  vacancies_count: 3,
  applications_count: 2,
  created_at: '2019-06-04 04:56:39',
  updated_at: '2019-06-04 04:56:39'
};
describe('Test Map Role cards', () => {
  it('should render correctly', () => {
    const props = {
      roleData: [role],
      fetchActiveEngineers: jest.fn(),
      loading: true,
      activeEngineers: {},
      type: 'role'
    };
    const wrapper = shallow(<MapProjectRoleCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render more than one role correctly', () => {
    const props = {
      roleData: [role, role],
      fetchActiveEngineers: jest.fn(),
      loading: true,
      activeEngineers: {},
      type: 'role'
    };
    const wrapper = shallow(<MapProjectRoleCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
