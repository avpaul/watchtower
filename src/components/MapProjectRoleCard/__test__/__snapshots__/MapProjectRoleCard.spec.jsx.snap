// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Map Role cards Should render more than one role correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapProjectRoleCard
    activeEngineers={Object {}}
    fetchActiveEngineers={[Function]}
    loading={true}
    roleData={
        Array [
            Object {
              "active_engineers_count": 1,
              "applications_count": 2,
              "created_at": "2019-06-04 04:56:39",
              "description": "test role description",
              "id": 1,
              "name": "test role",
              "updated_at": "2019-06-04 04:56:39",
              "vacancies_count": 3,
            },
            Object {
              "active_engineers_count": 1,
              "applications_count": 2,
              "created_at": "2019-06-04 04:56:39",
              "description": "test role description",
              "id": 1,
              "name": "test role",
              "updated_at": "2019-06-04 04:56:39",
              "vacancies_count": 3,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
>
        <div
                className="col-4 mb-4"
        >
                <RoleCard
                        activeEngineers={Object {}}
                        fetchActiveEngineers={[Function]}
                        loading={true}
                        role={
                                Object {
                                        "active_engineers_count": 1,
                                        "applications_count": 2,
                                        "created_at": "2019-06-04 04:56:39",
                                        "description": "test role description",
                                        "id": 1,
                                        "name": "test role",
                                        "updated_at": "2019-06-04 04:56:39",
                                        "vacancies_count": 3,
                                      }
                        }
                />
        </div>
        <div
                className="col-4 mb-4"
        >
                <RoleCard
                        activeEngineers={Object {}}
                        fetchActiveEngineers={[Function]}
                        loading={true}
                        role={
                                Object {
                                        "active_engineers_count": 1,
                                        "applications_count": 2,
                                        "created_at": "2019-06-04 04:56:39",
                                        "description": "test role description",
                                        "id": 1,
                                        "name": "test role",
                                        "updated_at": "2019-06-04 04:56:39",
                                        "vacancies_count": 3,
                                      }
                        }
                />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-4 mb-4"
>
            <RoleCard
                        activeEngineers={Object {}}
                        fetchActiveEngineers={[Function]}
                        loading={true}
                        role={
                                    Object {
                                                "active_engineers_count": 1,
                                                "applications_count": 2,
                                                "created_at": "2019-06-04 04:56:39",
                                                "description": "test role description",
                                                "id": 1,
                                                "name": "test role",
                                                "updated_at": "2019-06-04 04:56:39",
                                                "vacancies_count": 3,
                                              }
                        }
            />
</div>,
          <div
            className="col-4 mb-4"
>
            <RoleCard
                        activeEngineers={Object {}}
                        fetchActiveEngineers={[Function]}
                        loading={true}
                        role={
                                    Object {
                                                "active_engineers_count": 1,
                                                "applications_count": 2,
                                                "created_at": "2019-06-04 04:56:39",
                                                "description": "test role description",
                                                "id": 1,
                                                "name": "test role",
                                                "updated_at": "2019-06-04 04:56:39",
                                                "vacancies_count": 3,
                                              }
                        }
            />
</div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <RoleCard
              activeEngineers={Object {}}
              fetchActiveEngineers={[Function]}
              loading={true}
              role={
                            Object {
                                          "active_engineers_count": 1,
                                          "applications_count": 2,
                                          "created_at": "2019-06-04 04:56:39",
                                          "description": "test role description",
                                          "id": 1,
                                          "name": "test role",
                                          "updated_at": "2019-06-04 04:56:39",
                                          "vacancies_count": 3,
                                        }
              }
/>,
            "className": "col-4 mb-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeEngineers": Object {},
              "fetchActiveEngineers": [Function],
              "loading": true,
              "role": Object {
                "active_engineers_count": 1,
                "applications_count": 2,
                "created_at": "2019-06-04 04:56:39",
                "description": "test role description",
                "id": 1,
                "name": "test role",
                "updated_at": "2019-06-04 04:56:39",
                "vacancies_count": 3,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <RoleCard
              activeEngineers={Object {}}
              fetchActiveEngineers={[Function]}
              loading={true}
              role={
                            Object {
                                          "active_engineers_count": 1,
                                          "applications_count": 2,
                                          "created_at": "2019-06-04 04:56:39",
                                          "description": "test role description",
                                          "id": 1,
                                          "name": "test role",
                                          "updated_at": "2019-06-04 04:56:39",
                                          "vacancies_count": 3,
                                        }
              }
/>,
            "className": "col-4 mb-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeEngineers": Object {},
              "fetchActiveEngineers": [Function],
              "loading": true,
              "role": Object {
                "active_engineers_count": 1,
                "applications_count": 2,
                "created_at": "2019-06-04 04:56:39",
                "description": "test role description",
                "id": 1,
                "name": "test role",
                "updated_at": "2019-06-04 04:56:39",
                "vacancies_count": 3,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <div
                    className="col-4 mb-4"
          >
                    <RoleCard
                              activeEngineers={Object {}}
                              fetchActiveEngineers={[Function]}
                              loading={true}
                              role={
                                        Object {
                                                  "active_engineers_count": 1,
                                                  "applications_count": 2,
                                                  "created_at": "2019-06-04 04:56:39",
                                                  "description": "test role description",
                                                  "id": 1,
                                                  "name": "test role",
                                                  "updated_at": "2019-06-04 04:56:39",
                                                  "vacancies_count": 3,
                                                }
                              }
                    />
          </div>
          <div
                    className="col-4 mb-4"
          >
                    <RoleCard
                              activeEngineers={Object {}}
                              fetchActiveEngineers={[Function]}
                              loading={true}
                              role={
                                        Object {
                                                  "active_engineers_count": 1,
                                                  "applications_count": 2,
                                                  "created_at": "2019-06-04 04:56:39",
                                                  "description": "test role description",
                                                  "id": 1,
                                                  "name": "test role",
                                                  "updated_at": "2019-06-04 04:56:39",
                                                  "vacancies_count": 3,
                                                }
                              }
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-4 mb-4"
>
              <RoleCard
                            activeEngineers={Object {}}
                            fetchActiveEngineers={[Function]}
                            loading={true}
                            role={
                                          Object {
                                                        "active_engineers_count": 1,
                                                        "applications_count": 2,
                                                        "created_at": "2019-06-04 04:56:39",
                                                        "description": "test role description",
                                                        "id": 1,
                                                        "name": "test role",
                                                        "updated_at": "2019-06-04 04:56:39",
                                                        "vacancies_count": 3,
                                                      }
                            }
              />
</div>,
            <div
              className="col-4 mb-4"
>
              <RoleCard
                            activeEngineers={Object {}}
                            fetchActiveEngineers={[Function]}
                            loading={true}
                            role={
                                          Object {
                                                        "active_engineers_count": 1,
                                                        "applications_count": 2,
                                                        "created_at": "2019-06-04 04:56:39",
                                                        "description": "test role description",
                                                        "id": 1,
                                                        "name": "test role",
                                                        "updated_at": "2019-06-04 04:56:39",
                                                        "vacancies_count": 3,
                                                      }
                            }
              />
</div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <RoleCard
                activeEngineers={Object {}}
                fetchActiveEngineers={[Function]}
                loading={true}
                role={
                                Object {
                                                "active_engineers_count": 1,
                                                "applications_count": 2,
                                                "created_at": "2019-06-04 04:56:39",
                                                "description": "test role description",
                                                "id": 1,
                                                "name": "test role",
                                                "updated_at": "2019-06-04 04:56:39",
                                                "vacancies_count": 3,
                                              }
                }
/>,
              "className": "col-4 mb-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeEngineers": Object {},
                "fetchActiveEngineers": [Function],
                "loading": true,
                "role": Object {
                  "active_engineers_count": 1,
                  "applications_count": 2,
                  "created_at": "2019-06-04 04:56:39",
                  "description": "test role description",
                  "id": 1,
                  "name": "test role",
                  "updated_at": "2019-06-04 04:56:39",
                  "vacancies_count": 3,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <RoleCard
                activeEngineers={Object {}}
                fetchActiveEngineers={[Function]}
                loading={true}
                role={
                                Object {
                                                "active_engineers_count": 1,
                                                "applications_count": 2,
                                                "created_at": "2019-06-04 04:56:39",
                                                "description": "test role description",
                                                "id": 1,
                                                "name": "test role",
                                                "updated_at": "2019-06-04 04:56:39",
                                                "vacancies_count": 3,
                                              }
                }
/>,
              "className": "col-4 mb-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeEngineers": Object {},
                "fetchActiveEngineers": [Function],
                "loading": true,
                "role": Object {
                  "active_engineers_count": 1,
                  "applications_count": 2,
                  "created_at": "2019-06-04 04:56:39",
                  "description": "test role description",
                  "id": 1,
                  "name": "test role",
                  "updated_at": "2019-06-04 04:56:39",
                  "vacancies_count": 3,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Test Map Role cards should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapProjectRoleCard
    activeEngineers={Object {}}
    fetchActiveEngineers={[Function]}
    loading={true}
    roleData={
        Array [
            Object {
              "active_engineers_count": 1,
              "applications_count": 2,
              "created_at": "2019-06-04 04:56:39",
              "description": "test role description",
              "id": 1,
              "name": "test role",
              "updated_at": "2019-06-04 04:56:39",
              "vacancies_count": 3,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
>
        <div
                className="col-4 mb-4"
        >
                <RoleCard
                        activeEngineers={Object {}}
                        fetchActiveEngineers={[Function]}
                        loading={true}
                        role={
                                Object {
                                        "active_engineers_count": 1,
                                        "applications_count": 2,
                                        "created_at": "2019-06-04 04:56:39",
                                        "description": "test role description",
                                        "id": 1,
                                        "name": "test role",
                                        "updated_at": "2019-06-04 04:56:39",
                                        "vacancies_count": 3,
                                      }
                        }
                />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-4 mb-4"
>
            <RoleCard
                        activeEngineers={Object {}}
                        fetchActiveEngineers={[Function]}
                        loading={true}
                        role={
                                    Object {
                                                "active_engineers_count": 1,
                                                "applications_count": 2,
                                                "created_at": "2019-06-04 04:56:39",
                                                "description": "test role description",
                                                "id": 1,
                                                "name": "test role",
                                                "updated_at": "2019-06-04 04:56:39",
                                                "vacancies_count": 3,
                                              }
                        }
            />
</div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <RoleCard
              activeEngineers={Object {}}
              fetchActiveEngineers={[Function]}
              loading={true}
              role={
                            Object {
                                          "active_engineers_count": 1,
                                          "applications_count": 2,
                                          "created_at": "2019-06-04 04:56:39",
                                          "description": "test role description",
                                          "id": 1,
                                          "name": "test role",
                                          "updated_at": "2019-06-04 04:56:39",
                                          "vacancies_count": 3,
                                        }
              }
/>,
            "className": "col-4 mb-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeEngineers": Object {},
              "fetchActiveEngineers": [Function],
              "loading": true,
              "role": Object {
                "active_engineers_count": 1,
                "applications_count": 2,
                "created_at": "2019-06-04 04:56:39",
                "description": "test role description",
                "id": 1,
                "name": "test role",
                "updated_at": "2019-06-04 04:56:39",
                "vacancies_count": 3,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <div
                    className="col-4 mb-4"
          >
                    <RoleCard
                              activeEngineers={Object {}}
                              fetchActiveEngineers={[Function]}
                              loading={true}
                              role={
                                        Object {
                                                  "active_engineers_count": 1,
                                                  "applications_count": 2,
                                                  "created_at": "2019-06-04 04:56:39",
                                                  "description": "test role description",
                                                  "id": 1,
                                                  "name": "test role",
                                                  "updated_at": "2019-06-04 04:56:39",
                                                  "vacancies_count": 3,
                                                }
                              }
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-4 mb-4"
>
              <RoleCard
                            activeEngineers={Object {}}
                            fetchActiveEngineers={[Function]}
                            loading={true}
                            role={
                                          Object {
                                                        "active_engineers_count": 1,
                                                        "applications_count": 2,
                                                        "created_at": "2019-06-04 04:56:39",
                                                        "description": "test role description",
                                                        "id": 1,
                                                        "name": "test role",
                                                        "updated_at": "2019-06-04 04:56:39",
                                                        "vacancies_count": 3,
                                                      }
                            }
              />
</div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <RoleCard
                activeEngineers={Object {}}
                fetchActiveEngineers={[Function]}
                loading={true}
                role={
                                Object {
                                                "active_engineers_count": 1,
                                                "applications_count": 2,
                                                "created_at": "2019-06-04 04:56:39",
                                                "description": "test role description",
                                                "id": 1,
                                                "name": "test role",
                                                "updated_at": "2019-06-04 04:56:39",
                                                "vacancies_count": 3,
                                              }
                }
/>,
              "className": "col-4 mb-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeEngineers": Object {},
                "fetchActiveEngineers": [Function],
                "loading": true,
                "role": Object {
                  "active_engineers_count": 1,
                  "applications_count": 2,
                  "created_at": "2019-06-04 04:56:39",
                  "description": "test role description",
                  "id": 1,
                  "name": "test role",
                  "updated_at": "2019-06-04 04:56:39",
                  "vacancies_count": 3,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
